proposals/NNNN-reinstate-self-for-accessing-instance-properties-and-functions.md

```diff
@@ -0,0 +1,99 @@
+# Re-instate mandatory self for accessing instance properties and functions  
+
+* Proposal: [SE-NNNN](https://github.com/apple/swift-evolution/blob/master/proposals/NNNN-mandatory-self-for-accessing-instance-properties-and-functions.md)
+* Author(s): [David Hart](https://github.com/hartbit)
+* Status: **Review**
+* Review manager: TBD
+
+## Introduction
+
+Swift used to force using `self.` when accessing instance properties and functions, but since dropped this requirement (only keeping it for closure captures). It now feels inconsistent, and we have lost the automatic documentation of instance properties vs local variables and instance functions vs local functions or closures. This proposal offers to re-instate the original behaviour.
```

Comment:
That’s not true :) It was enabled early in the Swift betas and remove in Xcode 6.3 Beta 3:

A previous beta began requiring self. on the use of instance members within local functions (as it is for closures). This requirement was considered too broad and has been removed. (19945738)

> On 07 Dec 2015, at 05:22, Adam Christopher Smith <notifications@github.com> wrote:
> 
> In proposals/NNNN-reinstate-self-for-accessing-instance-properties-and-functions.md <https://github.com/apple/swift-evolution/pull/30#discussion_r46785784>:
> 
> > @@ -0,0 +1,99 @@
> > +# Re-instate mandatory self for accessing instance properties and functions  
> > +
> > +* Proposal: [SE-NNNN](https://github.com/apple/swift-evolution/blob/master/proposals/NNNN-mandatory-self-for-accessing-instance-properties-and-functions.md)
> > +* Author(s): [David Hart](https://github.com/hartbit)
> > +* Status: **Review**
> > +* Review manager: TBD
> > +
> > +## Introduction
> > +
> > +Swift used to force using `self.` when accessing instance properties and functions, but since dropped this requirement (only keeping it for closure captures). It now feels inconsistent, and we have lost the automatic documentation of instance properties vs local variables and instance functions vs local functions or closures. This proposal offers to re-instate the original behaviour.
> Since the very first developer beta, there has never been a requirement to reference instance variables or methods with self outside of a closure. I understand this is just a point of history, but "re-instating" isn't quite the correct term for this proposal.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/apple/swift-evolution/pull/30/files#r46785784>.
>